cosmpy/aerial/client/__init__.py

```diff
@@ -178,6 +238,169 @@ def send_tokens(
             self, tx, sender, gas_limit=gas_limit, memo=memo
         )
 
+    def query_validators(
+        self, status: Optional[ValidatorStatus] = None
+    ) -> List[Validator]:
+        filtered_status = status or ValidatorStatus.BONDED
+
+        req = QueryValidatorsRequest()
+        if filtered_status != ValidatorStatus.UNSPECIFIED:
+            req.status = filtered_status.value
+
+        resp = self.staking.Validators(req)
+
+        validators: List[Validator] = []
+        for validator in resp.validators:
+            validators.append(
+                Validator(
+                    address=Address(validator.operator_address),
+                    tokens=int(validator.tokens),
+                    moniker=str(validator.description.moniker),
+                    status=ValidatorStatus.from_proto(validator.status),
+                )
+            )
+        return validators
+
+    def query_staking_summary(self, address: Address) -> StakingSummary:
+        current_positions: List[StakingPosition] = []
+
+        req = QueryDelegatorDelegationsRequest(delegator_addr=str(address))
+        resp = self.staking.DelegatorDelegations(req)
+
+        # TODO: Need pagination support
+        for item in resp.delegation_responses:
+
+            req = QueryDelegationRewardsRequest(
+                delegator_address=str(address),
+                validator_address=str(item.delegation.validator_address),
+            )
+            rewards_resp = self.distribution.DelegationRewards(req)
+
+            stake_reward = 0
+            for reward in rewards_resp.rewards:
+                if reward.denom == self.network_config.staking_denomination:
+                    stake_reward = int(reward.amount) // (10**18)
```

Comment:
why `// (10**18)` here ? 
This is unlikely to work with any other tokens than FET and lose quite a bit of precision which might not be expected from the end-user POV?

Reply:
This is actually a Cosmos SDK thing as I understand it. It is because DecCoin is hard coded to 10**18